### PR TITLE
chore(RHTAPWATCH-886): smee deployment replicas alert rule

### PR DESCRIPTION
**Pull request by avi-biton**
- Create alert rule to monitor deployment replicas of smee
- Add tests to the alert

### CODE REVIEW

File: rhobs/alerting/data_plane/prometheus.smee_deployment_alerts.yaml [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+  name: rhtap-smee-alerting-rules
+  labels:
+    tenant: rhtap
+spec:
+  groups:
+  - name: smee-deployment-replicas-alerts
+    interval: 1m
+    rules:
+    - alert: SmeeDeploymentReplicasMismatch
+      expr: kube_deployment_spec_replicas{namespace=~"smee.*"} != kube_deployment_status_replicas_available{namespace=~"smee.*"}
```
Comment (yftacherzog) [inline rhobs/alerting/data_plane/prometheus.smee_deployment_alerts.yaml:13]: I don't see those time series in AppSRE Grafana. We need to figure out how come they are not there.

File: test/promql/tests/data_plane/mismatch_smee_deployment_replicas_test.yaml [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+      - eval_time: 5m
+        alertname: SmeeDeploymentReplicasMismatch
+        exp_alerts:
+          - exp_labels:
+              severity: warning
+              namespace: smee-client
+              deployment: gosmee
+              source_cluster: cluster01
+            exp_annotations:
+              summary: >-
```
[the remark this answers is not in the document]
Comment (avi-biton) [reply test/promql/tests/data_plane/mismatch_smee_deployment_replicas_test.yaml:26]: Added to the description

File: rhobs/alerting/data_plane/prometheus.smee_deployment_alerts.yaml (hunk 1 above)
Comment (avi-biton) [reply rhobs/alerting/data_plane/prometheus.smee_deployment_alerts.yaml:13]: Thats weird, I can see them:
https://grafana.app-sre.devshift.net/explore?orgId=1&left=%7B%22datasource%22:%22PDCCF087A30F0737B%22,%22queries%22:%5B%7B%22refId%22:%22A%22,%22datasource%22:%7B%22type%22:%22prometheus%22,%22uid%22:%22PDCCF087A30F0737B%22%7D,%22editorMode%22:%22code%22,%22expr%22:%22kube_deployment_spec_replicas%22,%22legendFormat%22:%22__auto%22,%22range%22:true,%22instant%22:true%7D,%7B%22refId%22:%22B%22,%22datasource%22:%7B%22type%22:%22prometheus%22,%22uid%22:%22PDCCF087A30F0737B%22%7D,%22editorMode%22:%22code%22,%22expr%22:%22kube_deployment_status_replicas_available%22,%22legendFormat%22:%22__auto%22,%22range%22:true,%22instant%22:true%7D%5D,%22range%22:%7B%22from%22:%22now-5m%22,%22to%22:%22now%22%7D%7D

Comment (avi-biton) [reply rhobs/alerting/data_plane/prometheus.smee_deployment_alerts.yaml:13]: I get the `kube_deployment_spec_replicas`
https://grafana.app-sre.devshift.net/explore?orgId=1&left=%7B%22datasource%22:%22PDCCF087A30F0737B%22,%22queries%22:%5B%7B%22refId%22:%22A%22,%22datasource%22:%7B%22type%22:%22prometheus%22,%22uid%22:%22PDCCF087A30F0737B%22%7D,%22editorMode%22:%22code%22,%22expr%22:%22kube_deployment_spec_replicas%22,%22legendFormat%22:%22__auto%22,%22range%22:true,%22instant%22:true%7D%5D,%22range%22:%7B%22from%22:%22now-5m%22,%22to%22:%22now%22%7D%7D

But I am not getting the `kube_deployment_status_replicas_available`

Comment (avi-biton) [reply rhobs/alerting/data_plane/prometheus.smee_deployment_alerts.yaml:13]: OK, it should be added to the service monitor:
https://github.com/redhat-appstudio/infra-deployments/blob/main/components/monitoring/prometheus/base/monitoringstack/monitoringstack.yaml#L111

Comment (yftacherzog) [reply rhobs/alerting/data_plane/prometheus.smee_deployment_alerts.yaml:13]: oh yes :)

**Review by yftacherzog — APPROVED** [review]
lgtm

**Pull request merged**